src/stan/lang/compiler.hpp

```diff
@@ -28,10 +29,11 @@ namespace stan {
     bool compile(std::ostream* msgs,
                  std::istream& stan_lang_in,
                  std::ostream& cpp_out,
-                 const std::string& model_name) {
+                 const std::string& model_name,
+                 const bool allow_undefined = false) {
```

Comment:
The default's good because it'll mean the `compile()` function will be backward compatible.